src/coreComponents/constitutive/relativePermeability/RelativePermeabilityBase.cpp

```diff
@@ -90,6 +93,9 @@ void RelativePermeabilityBase::resizeFields( localIndex const size, localIndex c
 
   m_phaseRelPerm.resize( size, numPts, numPhases );
   m_dPhaseRelPerm_dPhaseVolFrac.resize( size, numPts, numPhases, numPhases );
+  //phase trapped for stats
+  m_phaseTrapped.resize( size, numPhases );//2d(*) or 3d ??
+  m_phaseTrapped.setValues< parallelDevicePolicy<> >( 0.0 );
```

Comment:
```suggestion
  // phase trapped volume fraction for stats
  m_phaseTrappedVolFrac.resize( size, numPhases );
  m_phaseTrappedVolFrac.zero();
```
and I fully agree with your choice of `array2d` (the same is done for `phaseMinHistoricalVolFrac`).

In LvArray, I found this comment from Ben in the Doxygen of the `zero` function:
```
This is preferred over setValues< POLICY >( 0 ) for numeric types since it is much faster in most cases.
```
so I thought we could use it here.

Reply:
I though turn it to ``array3d`` as it is interesting for checking purpose to vtk-output and ``array2d`` seems to be interpreted as a scalar field (tested also on ``phaseMaxHistoricalVolFrac``).

Reply:
When plotting material data, VTK output interprets second dimension as quadrature point and collapses (averages over) it. Perhaps we need a more discriminate way of marking fields as having or not having a quadrature dimension. But for now you can just keep a dummy second dimension of size 1.
```c++
m_phaseTrappedVolFrac.resize( size, 1, numPhases );
```

Reply:
Newbie question, how come 
https://github.com/GEOSX/GEOSX/blob/ffe51866a1caa3089699f84d940c35241f6a72ad/src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBaseExtrinsicData.hpp#L82-L88
produces the correct output ?

Is it because of the diff between
https://github.com/GEOSX/GEOSX/blob/88c88fd7d51cde0f0fb6d0deaa6470a8b56efa32/src/coreComponents/constitutive/relativePermeability/RelativePermeabilityBase.cpp#L47

and

https://github.com/GEOSX/GEOSX/blob/ffe51866a1caa3089699f84d940c35241f6a72ad/src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp#L221-L223